job-service/src/main/java/com/hpe/caf/services/job/queue/QueueServices.java

```diff
@@ -124,8 +128,20 @@ public void sendMessage(
         }
 
         //  Send the message.
+        try {
+            publishMessage(taskMessageBytes);
+        } catch (IOException | TimeoutException e) {
+            LOG.warn("Failed to publish message. Will retry", e);
+            publishMessage(taskMessageBytes);
+        }
+    }
+
+    private void publishMessage(final byte[] taskMessageBytes)
+            throws IOException, InterruptedException, TimeoutException
+    {
         publisherChannel.basicPublish(
                 "", targetQueue, MessageProperties.TEXT_PLAIN, taskMessageBytes);
```

Comment:
Should this also be `MessageProperties.PERSISTENT_TEXT_PLAIN`?

Reply:
Yes probably, assuming this wouldn't cause disk issues. (Also worth noting no of the works send with `PERSISTENT_TEXT_PLAIN`)

Reply:
Really, what are the implications?  Do we need to raise a Jira for it?

Reply:
So the persistent flag means messages will be written to disk immediately on arrival to the queue regardless of current memory levels. This means on durable queues any persistent messages will be recovered if the node restarts as opposed to transient messages which are only ever stored in memory (unless near overflow) and are lost of the node goes down.

This is all separate from mirrored queues (which we have in prod) and only really applies to single node recovery. It only comes into play if multiple nodes die before the messages can be replicated. (WHich seems very unlikely)

Reply:
Ok.  I assume the acknowledgement is never received until the message has been secured.  If so then it sounds like it's not an issue.